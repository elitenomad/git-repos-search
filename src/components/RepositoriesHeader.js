import React from "react"
import {Container, Row, Col} from "react-bootstrap"
import SimplePagination from './Pagination/SimplePagination'

const RepositoriesHeader = (props) => {
  const {repoCount, handleSubmit} = props
  const pages = Math.ceil(repoCount / 30) // 30 is the default results returns by Github API.
  
  return (
    <Container fluid>
      <Row>
        <Col xs={6} className="text-left">
          <p>Found {repoCount} repositories.</p>
        </Col>
        <Col xs={6} className="text-right">
            <SimplePagination
              pages={pages}
              onClick={handleSubmit}
            />
        </Col>
      </Row>
    </Container>
  )
}

export default RepositoriesHeader
