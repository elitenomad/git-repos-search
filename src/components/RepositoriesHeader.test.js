import React from "react"
import renderer from "react-test-renderer"
import RepositoriesHeader from "./RepositoriesHeader"

describe("RepositoriesHeader", () => {
  test("for given props matches snapshot", () => {
    const props = {
      repoCount: 125,
    }
    const tree = renderer.create(<RepositoriesHeader {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
