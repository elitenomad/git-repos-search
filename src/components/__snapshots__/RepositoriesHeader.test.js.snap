// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesHeader for given props matches snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="text-left col-6"
    >
      <p>
        Found 
        125
         repositories.
      </p>
    </div>
    <div
      className="text-right col-6"
    >
      <ul
        className="pagination-end pagination"
      >
        <li
          className="page-item disabled"
        >
          <span
            className="page-link"
            disabled={true}
            onClick={[Function]}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="sr-only"
            >
              Previous
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="sr-only"
            >
              Next
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
